File: gto-support-jsonapi/src/test/java/org/ccci/gto/android/common/jsonapi/converter/LocaleTypeConverterTest.java
package org.ccci.gto.android.common.jsonapi.converter;

import org.junit.Test;

import java.io.Serializable;
import java.util.Locale;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.nullValue;
import static org.junit.Assert.assertThat;

public class LocaleTypeConverterTest {
    @Test
    public void verifySupports() throws Exception {
        final LocaleTypeConverter converter = LocaleTypeConverter.INSTANCE;
        assertThat(converter.supports(Object.class), is(false));
        assertThat(converter.supports(Locale.class), is(true));
        assertThat(converter.supports(Cloneable.class), is(false));
        assertThat(converter.supports(Serializable.class), is(false));
    }

    @Test
    public void verifyToString() throws Exception {
        final LocaleTypeConverter converter = LocaleTypeConverter.INSTANCE;

        assertThat(converter.toString(Locale.ENGLISH), is("en"));
        assertThat(converter.toString(Locale.CANADA), is("en-CA"));
    }

    @Test
    public void verifyFromString() throws Exception {
        final LocaleTypeConverter converter = LocaleTypeConverter.INSTANCE;

        assertThat(converter.fromString("en"), is(Locale.ENGLISH));
        assertThat(converter.fromString("en-CA"), is(Locale.CANADA));
    }

    @Test
    public void verifyNull() throws Exception {
        final LocaleTypeConverter converter = LocaleTypeConverter.INSTANCE;

        assertThat(converter.fromString(null), nullValue());
        assertThat(converter.toString(null), nullValue());
    }
}

File: gto-support-jsonapi/src/test/kotlin/org/ccci/gto/android/common/jsonapi/converter/LocaleTypeConverterTest.kt
package org.ccci.gto.android.common.jsonapi.converter

import java.io.Serializable
import java.util.Locale
import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.CoreMatchers.nullValue
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Test

class LocaleTypeConverterTest {
    private val converter = LocaleTypeConverter

    @Test
    fun verifySupports() {
        assertThat(converter.supports(Any::class.java), equalTo(false))
        assertThat(converter.supports(Locale::class.java), equalTo(true))
        assertThat(converter.supports(Cloneable::class.java), equalTo(false))
        assertThat(converter.supports(Serializable::class.java), equalTo(false))
    }

    @Test
    fun verifyToString() {
        assertThat(converter.toString(Locale.ENGLISH), equalTo("en"))
        assertThat(converter.toString(Locale.CANADA), equalTo("en-CA"))
    }

    @Test
    fun verifyFromString() {
        assertThat(converter.fromString("en"), equalTo(Locale.ENGLISH))
        assertThat(converter.fromString("en-CA"), equalTo(Locale.CANADA))
    }

    @Test
    fun verifyNull() {
        assertThat(converter.fromString(null), nullValue())
        assertThat(converter.toString(null), nullValue())
    }
}
